clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -1077,4 +1131,18 @@ public void onUpdate(ClusterResource clusterResource) {
             reconcile();
         }
     }
+
+    /**
+     * Register a new listener that will be invoked whenever the member state changes, or a new
+     * member ID or epoch is received.
+     *
+     * @param listener Listener to invoke.
+     */
+    @Override
+    public void registerStateListener(MemberStateListener listener) {
+        if (listener == null) {
+            throw new IllegalArgumentException("State updates listener cannot be null");
+        }
+        this.stateUpdatesListeners.add(listener);
+    }
```

Comment:
Do we need the ability to _remove_ a `MemberStateListener` from the set of listeners?

Reply:
I'd say YAGNI

Reply:
Agree on YAGNI. It's mainly to let the commit mgr access the latest member id/epoch from the state machine, avoiding a circular dependency, and ensuring that the state machine is the only one updating those and passing them on read-only.